de.metas.business/src/main/java/de/metas/purchasing/api/IBPartnerProductDAO.java

```diff
@@ -40,6 +41,9 @@
  */
 public interface IBPartnerProductDAO extends ISingletonService
 {
+
+	public static final String MSG_ProductSalesExclusionError = "ProductSalesExclusionError";
```

Comment:
Pls move it to BPartnerProductBL because that's the only place where is used.
There is no need to be exposed in interface.

Reply:
done